Automattic/wp-calypso
# Me: Improve "Change Username" Account Settings

Fixes #94464; fixes #94465

## Proposed Changes

* Improves the "Change username" settings under Me > Account Settings

## Why are these changes being made?

See original issue - this section currently feels 'off' because it doesn't match the Calypso design guidelines. I'd argue that the `Notice` component should really be used to emphasise the important information (ie. the actual warning), and that `FormInputValidation` should inform the user whether their username is valid. 

## Testing Instructions

Navigate to Me > Account Settings and try changing your username. Verify that you still need to confirm your username for this change to work, and that you are still warned about the consequences for this change.

| Current | Proposed |
|--------|--------|
| <img width="765" alt="Screenshot 2024-09-17 at 20 28 41" src="https://github.com/user-attachments/assets/302603d9-a834-4bcc-a1fd-c9199923d71a"> | <img width="967" alt="Screenshot 2024-09-17 at 20 28 46" src="https://github.com/user-attachments/assets/b0d83ff6-903f-44ef-a111-c9374a95fd25"> | 

cc @taipeicoder, @okmttdhr, @m1r0
